lib/rack/cors/resources.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require_relative 'resources/cors_misconfiguration_error'
+
+module Rack
+  class Cors
+    class Resources
+      attr_reader :resources
+
+      def initialize
+        @origins = []
+        @resources = []
+        @public_resources = false
+      end
+
+      def origins(*args, &blk)
+        @origins = args.flatten.reject { |s| s == '' }.map do |n|
+          case n
+          when Proc, Regexp, %r{^https?://}, 'file://'
+            n
+          when '*'
+            @public_resources = true
+            n
+          else
+            Regexp.compile("^[a-z][a-z0-9.+-]*:\\\/\\\/#{Regexp.quote(n)}$")
+          end
+        end.flatten
+        @origins.push(blk) if blk
+      end
+
+      def resource(path, opts = {})
+        @resources << Resource.new(public_resources?, path, opts)
+      end
+
+      def public_resources?
+        @public_resources
+      end
+
+      def allow_origin?(source, env = {})
+        return true if public_resources?
+
+        !!@origins.detect do |origin|
+          if origin.is_a?(Proc)
+            origin.call(source, env)
+          elsif origin.is_a?(Regexp)
+            source =~ origin
+          else
+            source == origin
+          end
```

Comment:
I don't believe it is, since everything other than a `Proc`, `Regexp` or `String("*")` gets compiled into a `Regexp` here: https://github.com/cyu/rack-cors/blob/e8e91ae21e79c245af83ba9d16156e2326812ebd/lib/rack/cors/resources.rb#L25

Reply:
ah, okay.